torch/csrc/jit/autodiff.cpp

```diff
@@ -267,6 +278,15 @@ struct ReverseDetails {
   Block * reverse_block;
 };
 
+// AutogradAdd is a special addition function that handles Undef
+// AutogradAdd(a, b) == a + b if defined(a) and defined(b)
+// AutogradAdd(Undef, b) == b
+// AutogradAdd(a, Undef) == a
```

Comment:
AutogradAdd(Undef, Undef) == Undef?

Reply:
Yes, either of the above cases match that.